### DISCUSSION TITLE
import python libraries with argo workflow script

### DISCUSSION THREAD

**Discussion by sreeragu** [argoproj/argo-workflows#7833] [Q&A] [answered]
Hi,
I'm testing a Workflow with a sensor. I'm getting a json input from the sensor and the workflow contains a python script which process this input json. I need to use custom python modules like 'requests', 'yaml' etc.

---
                templates:
                - name: my-script
                  inputs:
                    parameters:
                    - name: json_data
                  script:
                    image: python:alpine3.6
                    command: [python]
                    source: |
                      import json
                      import sys
                      import requests
                      import yaml
                      def somerequests():
                          requests.get('http://someendpoint.com/info')
                      if __name__ == "__main__":
                          input_json = {{inputs.parameters.json_data}}
---

I'm getting module import error.

---
W
Traceback (most recent call last):
  File "/argo/staging/script", line 3, in <module>
    import requests
ModuleNotFoundError: No module named 'requests'
---
I would like to understand how we can import / install these python modules to this script.

**Comment by terrytangyuan** [comment]
You can use container template instead and add command line flags  for your Python script. The parameters can then be passed as flags to the Python script.

**Reply by sreeragu** [reply]
Thanks for the info @terrytangyuan

**Comment by sreeragu** [comment, marked as answer]
I have fixed it by rebuilding the docker image for python alpine linux with the packages requests and pyyaml.
